src/react-apps/applications/runtime/src/components/base/FileUploadComponent.tsx

```diff
@@ -1,45 +1,332 @@
+                        />
+                      </div>
+                    }
+                  </td>
+                  <td>
+                    <div onClick={this.handleDeleteFile.bind(this, index)} id={'attachment-delete-' + index}>
+                      {!attachment.deleting &&
+                        <>
+                          {getLanguageFromKey('form_filler.file_uploader_list_delete', this.props.language)}
+                          <i className='ai ai-trash' />
+                        </>
+                      }
+                      {attachment.deleting &&
+                        <div className='a-loader' id={'loader-delete'}>
+                          <div
+                            className='loader loader-ellipsis'
+                            style={{ marginBottom: '1.6rem', marginRight: '1.0rem' }}
+                          />
+                        </div>
+                      }
+                    </div>
+                  </td>
+                </tr>
+              );
+            })}
+          </tbody>
+        </table>
+      </div>
+    );
+  }
+
+  public renderFileUploadContent = (): JSX.Element => {
+    const { hasCustomFileEndings, validFileEndings } = this.props.component;
+    return (
+      <div className={'container'}>
+        <div className='col text-center icon' style={{ marginTop: '3.5rem' }} >
           <i className='ai ai-upload' />
-        </label>
+        </div>
+        <div className='col text-center'>
+          <p className={'file-upload-text-bold'}>
+            {getLanguageFromKey('form_filler.file_uploader_drag', this.props.language)}
+            <span className={'file-upload-text-bold blue-underline'}>
+              {' ' + getLanguageFromKey('form_filler.file_uploader_find', this.props.language)}
+            </span>
+          </p>
+        </div>
+        <div className='col text-center'>
+          <p className={'file-upload-text'}>
+            {getLanguageFromKey('form_filler.file_uploader_valid_file_format', this.props.language)}
+            {hasCustomFileEndings ? (' ' + validFileEndings) :
+              (' ' + getLanguageFromKey('form_filler.file_upload_valid_file_format_all', this.props.language))}
+          </p>
+        </div>
       </div>
     );
   }
+
+  public renderAddMoreAttachmentsButton = (): JSX.Element => {
+    const { displayMode, maxNumberOfAttachments } = this.props.component;
+    if (displayMode === 'simple' && !this.state.showFileUpload &&
+      (this.state.attachments.length < maxNumberOfAttachments) && this.state.attachments.length > 0) {
+      return (
+        <button className={'file-upload-button blue-underline'} onClick={this.handleAddMoreAttachments}>
+          {getLanguageFromKey('form_filler.file_uploader_add_attachment', this.props.language)}
+        </button>
+      );
+    } else {
+      return null;
+    }
+  }
+
+  public handleAddMoreAttachments = () => {
```

Comment:
Confusing naming of function. This function doesn't handle anything, it just updates state.

Reply:
Renamed.